go.mod

```diff
@@ -1,6 +1,6 @@
 module go.opentelemetry.io/auto
 
-go 1.22.7
+go 1.23.2
```

Comment:
Hmm, this might be problematic. This would violate our compatibility policy: https://github.com/open-telemetry/opentelemetry-go-instrumentation?tab=readme-ov-file#compatibility

Reply:
I agree. this looks quite aggressive. 
For reference, when I wanted to update the dependency of this repo I had to bump the go.mod to 1.22.7 which is what we require today (https://github.com/odigos-io/odigos/pull/1776/commits/fb167f82c1f76947ee2f1d2966c5bc363a279fd1), and we have quite a lot of other dependencies and none of them required this relatively new version.

Reply:
Reverting back to the other library, since it doesn't seem like vitess will work for us right now :(